base/common/python/pki/nssdb.py

```diff
@@ -506,13 +506,18 @@ def find_keys(self, nickname=None, token=None):
             '-d', self.directory
         ]
 
-        if self.password_conf:
+        token = self.get_effective_token(token)
+
+        if token:
+            password_file = self.get_password_file(self.tmpdir, token)
+            cmd.extend(['-C', password_file])
+
+        elif self.password_conf:
             cmd.extend(['-f', self.password_conf])
 
         elif self.password_file:
             cmd.extend(['-C', self.password_file])
```

Comment:
Hmm.. I was assuming that the `password_conf` would provide the passwords for all tokens, so if it's available it should be used first. Is that not the case here?

Will this code work?
```
if self.password_conf:
    cmd.extend(['-f', self.password_conf])

elif token:
    password_file = self.get_password_file(self.tmpdir, token)
    cmd.extend(['-C', password_file])

elif self.password_file:
    cmd.extend(['-C', self.password_file])
```

Reply:
Yes, that's a good idea. I'll get it done.